### PR TITLE
Add initial support for Gaisler-BCC with Sparc

### PR DESCRIPTION
**Pull request by LinuxJedi**
# Description

Slight modifications and documentation to get wolfSSL working with Gaisler Sparc CPUs and their cross-compilers.

# Testing

Using TSIM3 LEON3 evaluation version (https://www.gaisler.com/index.php/downloads/simulators).

# Checklist

 - [ ] added tests
 - [ ] updated/added doxygen
 - [ ] updated appropriate READMEs
 - [ ] Updated manual and documentation

### CODE REVIEW

**Comment by douzzer** [comment]
retest this please

**Review by douzzer — CHANGES_REQUESTED** [review]
from `PRB-multi-test-script`:
```
[check-source-text] [2 of 7] [wolfssl]
    autogen.sh wolfssl...   real 0m14.913s  user 0m8.756s  sys 0m0.214s
    configure...   real 0m21.143s  user 0m9.405s  sys 0m2.425s
unrecognized macros used:
__sparc
add well-formed but unknown macros to /tmp/workspace/PRB-multi-test-script/wolfssl/.wolfssl_known_macro_extras.
```
(it means `.wolfssl_known_macro_extras` at the top of the repo.)

There's also an error in `PRB-fips-read-config` but I think it's a false positive.

**Comment by LinuxJedi** [comment]
Dang, when it kept failing I thought that might be the case, but I didn't see it in the Jenkins output. I'll fix ASAP.

**Comment by LinuxJedi** [comment]
@douzzer I fixed that problem. Any idea why Jenkins is still failing? I don't see it in the log.

**Comment by douzzer** [comment]
retest this please (`PRB-140-3 -tests` `140-3-known-tests/AES-GMAC_474643.json` "DRBG Continuous Test FIPS error")

**Comment by douzzer** [comment]
By the way, glibc no longer has support for pre-v8 sparc, and trying to build a toolchain for `sparc-unknown-linux-gnu` (to get a 32 bit sparc runtime) makes it think pre-v8 is targeted, so the build craps out.

TL;DR: Currently we have sparc64 testing in the nightly (just basic `testsuite.test`), but not sparc32.

P.s. interestingly, on sparc64, we have to disable sakke to get it to pass, otherwise `SEGV`.  Before this PR, also had to force on `WOLFSSL_SP_INT_DIGIT_ALIGN`.

**Review by douzzer — APPROVED** [review]
